### ISSUE TITLE
Docs: load() and image access object

### ISSUE THREAD

**Issue by gwideman** [python-pillow/Pillow#969]
Given that one of the most fundamental operations one would want to do on an image is to interact with its pixels, the current reference is very opaque on that subject, and could be vastly improved. In short, somewhere this should be described:

``` python
pixels = img.load()
p = pixels[px, py]
...
pixels[px, py] = p2
```

On page: http://pillow.readthedocs.org/reference/Image.html, the doc for Image.load() suggests that one rarely wants to call load(), and gives no explanation of why one _would_ want to call it. It mentions an "image access object", but there's no link, and that phrase returns nothing from the search function (not even this mention).

Yet from places like: http://stackoverflow.com/questions/2181292/using-the-image-point-method-in-pil-to-manipulate-pixel-data
... we learn that this "image access object" is pretty crucial for accessing the pixels. However the current docs for getpixel() and putpixel() don't mention it, though do say that the get/put functions are slow.

Bottom line: load() should document that it returns a PixelAccess object, and that object should be documented, or load()'s doc should make note of the [] method (and whether it's speedy and recommended). And getpixel() and putpixel() should mention it too.
